qiskit_experiments/curve_analysis/fit_models.py

```diff
@@ -0,0 +1,265 @@
+    ):
+        """Create new fit model.
+
+        Args:
+            fit_functions: List of callables that forms the fit model for a
+                particular curve analysis class. It may consists of multiple curves.
+            signatures: List of argument names that each fit function callable takes.
+                The length of the list should be identical to the ``fit_functions``.
+            fit_models: String representation of fit functions.
+                Because this is just a metadata, the format of input value doesn't matter.
+                It may be a single string description for the entire fit model, or
+                list of descriptions for each fit functions. If not provided,
+                "not defined" is stored in the experiment result metadata.
+            fixed_parameters: List of parameter names that are not considered to be fit parameter.
+                The value of parameter is provided by analysis default setting or users,
+                which is fixed during the curve fitting. Arbitrary number of parameters
+                in the fit model can be fixed, however, every parameter should be
+                defined in the model.
+
+        Raises:
+            AnalysisError: When ``fit_functions`` and ``signatures`` don't match.
+        """
+        if len(fit_functions) != len(signatures):
+            raise AnalysisError("Different numbers of fit_functions and signatures are given.")
+
+        self._fit_functions = fit_functions
+        self._signatures = signatures
+
+        # String representation of the fit model. This is stored as a list of string.
+        if not fit_models or isinstance(fit_models, str):
+            fit_models = [fit_models]
+        self._fit_models = fit_models
+
+        # No validation is performed since this class is always instantiated from the
+        # curve analysis class itself. The validation is performed there.
+        if not fixed_parameters:
+            fixed_parameters = []
+        self._fixed_params = {p: None for p in fixed_parameters}
+
+        # Create signature of this fit model, i.e. this will be signature of scipy fit function.
+        # The curves comprising this model may have different signature.
+        # The signature of this fit model is union of parameters in all curves.
+        union_params = []
+        for signature in signatures:
+            for parameter in signature:
+                if parameter not in union_params and parameter not in fixed_parameters:
+                    union_params.append(parameter)
+        self._uniton_params = union_params
+
+    @abstractmethod
+    def __call__(self, x: np.ndarray, *params) -> np.ndarray:
+        pass
+
+    def bind_parameters(self, **kwparams):
+        """Assign values to the fixed parameters."""
+        bind_dict = {k: kwparams[k] for k in self._fixed_params.keys() if k in kwparams}
+        self._fixed_params.update(bind_dict)
```

Comment:
This is still needed because user may update it via the set option after instance is initialized. If you don't want to allow this, fixed parameter dict should be in the class attribute rather than the default options.